libminifi/test/unit/SchedulingAgentTests.cpp

```diff
@@ -36,133 +38,166 @@ class CountOnTriggersProcessor : public minifi::core::Processor {
+  CHECK(first_task_reschedule_info.getNextExecutionTime() < std::chrono::steady_clock::now());
+  auto count_num_after_one_schedule = count_proc_->getNumberOfTriggers();
+  CHECK(count_num_after_one_schedule > 100);
+
+  auto second_task_reschedule_info = event_driven_agent->run(count_proc_.get(), context_, factory_);
+  CHECK(!second_task_reschedule_info.isFinished());
+  CHECK(second_task_reschedule_info.getNextExecutionTime() < std::chrono::steady_clock::now());
+  auto count_num_after_two_schedule = count_proc_->getNumberOfTriggers();
+  CHECK(count_num_after_two_schedule > count_num_after_one_schedule+100);
+}
+
+TEST_CASE_METHOD(SchedulingAgentTestFixture, "Cron Driven every year") {
+  count_proc_->setCronPeriod("0 0 0 1 1 ?");
+  auto cron_driven_agent = std::make_shared<CronDrivenSchedulingAgent>(gsl::make_not_null(controller_services_provider_.get()), test_repo_, test_repo_, content_repo_, configuration_, thread_pool_);
+  cron_driven_agent->start();
+  auto first_task_reschedule_info = cron_driven_agent->run(count_proc_.get(), context_, factory_);
+  CHECK(!first_task_reschedule_info.isFinished());
+  if (first_task_reschedule_info.getNextExecutionTime() > std::chrono::steady_clock::now() + 1min) {  // To avoid possibly failing around dec 31 23:59:59
+    auto wait_time_till_next_execution_time = std::chrono::round<std::chrono::seconds>(first_task_reschedule_info.getNextExecutionTime() - std::chrono::steady_clock::now());
+
+    auto current_time = date::make_zoned<std::chrono::seconds>(date::current_zone(), std::chrono::time_point_cast<std::chrono::seconds>(std::chrono::system_clock::now()));
+    auto current_year_month_day = date::year_month_day(date::floor<date::days>(current_time.get_local_time()));
+    auto new_years_day = date::make_zoned<std::chrono::seconds>(date::current_zone(), date::local_days{date::year{current_year_month_day.year()+date::years(1)}/date::January/1});
+
+    auto time_until_new_years_day = new_years_day.get_local_time() - current_time.get_local_time();
 
-  SECTION("Cron Driven every sec") {
-    count_proc->setCronPeriod("* * * * * *");
-    auto cron_driven_agent = std::make_shared<CronDrivenSchedulingAgent>(gsl::make_not_null(controller_services_provider_.get()), test_repo, test_repo, content_repo, configuration, thread_pool);
-    cron_driven_agent->start();
-    auto first_task_reschedule_info = cron_driven_agent->run(count_proc.get(), context, factory);
-    CHECK(!first_task_reschedule_info.isFinished());
-    CHECK(first_task_reschedule_info.getNextExecutionTime() <= std::chrono::steady_clock::now() + 1s);
-    CHECK(count_proc->getNumberOfTriggers() == 0);
-
-    std::this_thread::sleep_until(first_task_reschedule_info.getNextExecutionTime());
-    auto second_task_reschedule_info = cron_driven_agent->run(count_proc.get(), context, factory);
+    CHECK(std::chrono::round<std::chrono::minutes>(time_until_new_years_day - wait_time_till_next_execution_time) == 0min);
```

Comment:
I'm afraid that this is going to rarely fail when we go over the minute mark between the two timestamps, so I'd do a check for absolute difference being smaller than epsilon. (e.g. 1 min)